google/validation.go

```diff
@@ -6,6 +6,8 @@ import (
 	"regexp"
 )
 
+const ProjectRegex = "(?:(?:[-a-z0-9]{1,63}\\.)*(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?):)?(?:[0-9]{1,19}|(?:[a-z0-9](?:[-a-z0-9]{0,61}[a-z0-9])?))"
```

Comment:
Project IDs generated through Terraform will follow [this form](https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects#Project.FIELDS.project_id), but I'm not sure if those rules have been followed for the entire lifetime of project ids in general.

We should be able to match against `[^/]*` and capture anything inside the projects/ part of a self link without overmatching past a `/`.